arch/risc-v/src/common/riscv_doirq.c

```diff
@@ -71,9 +71,12 @@ uintreg_t *riscv_doirq(int irq, uintreg_t *regs)
   if (irq >= RISCV_IRQ_ECALLU && irq <= RISCV_IRQ_ECALLM)
     {
       regs[REG_EPC] += 4;
+      if (regs[REG_A0] != SYS_restore_context)
```

Comment:
Looks a bit messy, is this again because we use a system call to do the context restore ? I have always been wondering why waste the full exception handling path to do something this simple.

Reply:
Actually, restoring context doesn't necessarily require entering the exception handling mechanism.
Perhaps it's to achieve a unified implementation across different archs. We may optimize this part later.

Reply:
Yes, restoring context needs:
- Change address environment
- Return from exception
- (sched_note stuff?)

And it can of course be optimized later.